src/compiler.ts

```diff
@@ -9217,12 +9280,394 @@ export class Compiler extends DiagnosticEmitter {
+        if (ifunc.is(CommonFlags.SET)) {
+          assert(ifunc.signature.parameterTypes.length == 1);
+          if (ifunc.signature.parameterTypes[0] != type) {
+            return module.nop();
+          }
+          let valueExpr = module.local_get(1, nativeType);
+          return module.store(
+            type.byteSize,
+            thisExpr,
+            valueExpr,
+            nativeType,
+            field.memoryOffset
+           );
+        }
+        if (ifunc.is(CommonFlags.GET)) {
+          if (ifunc.signature.returnType != type) {
+            return module.nop();
+          }
+          assert(field.memoryOffset >= 0);
+          return this.module.load(
+            field.type.byteSize,
+            field.type.is(TypeFlags.SIGNED | TypeFlags.INTEGER),
+            thisExpr,
+            field.type.toNativeType(),
+            field.memoryOffset);
+        }
+      }
+      case ElementKind.PROPERTY: {
+        let p = <Property> prop;
+        if (ifunc.is(CommonFlags.SET)) {
+          method = p.setterInstance!;
+        }else if (ifunc.is(CommonFlags.GET)) {
+          method = p.getterInstance!;
+        }else {
+          throw Error("Interface method " + ifunc.name + " should be a property");
+        }
+        break;
+
+      }
+      default: {
+        throw Error("Class " + ifunc.parent.name + " must have property " + prop.name);
+      }
+    }
+    var origMethodName = method.internalName;
+    /**
+     * If method is not virtual it must be renamed to not overlap with actual implementation.
+     */
+    if (!method.is(CommonFlags.VIRTUAL) && ifunc.parent == method.parent) {
+      method.internalName = method.internalName + "~virtual";
+      method.unset(CommonFlags.COMPILED);
+    }
+
+    this.compileFunction(method);
+    const locals: usize[] = [];
+    for (let i: i32 = 0; i < ifunc.localsByIndex.length; i++) {
+      const local = ifunc.localsByIndex[i];
+      locals.push(this.module.local_get(local.index, local.type.toNativeType()));
+    }
+    const callExpr = this.module.call(method.internalName, locals, returnType);
+    method.internalName = origMethodName;
+    return callExpr;
+
+  }
+
+  loadClassArg(index: i32): ExpressionRef {
+    return this.module.local_get(index, this.options.usizeType.toNativeType());
+  }
+
+  get nativeUsizeType(): NativeType {
+    return this.options.usizeType.toNativeType();
+  }
+
+  loadClassID(): ExpressionRef {
+    var module = this.module;
+    return module.load(
+      4,
+      false,
+      module.binary(
+        BinaryOp.SubI32,
+        this.loadClassArg(0),
+        module.i32(8)
+      ),
+      NativeType.I32
+    );
+  }
 }
 
 // helpers
 
 const v128_zero = new Uint8Array(16);
 
+/* Useful for adding to the end of a block that needs a type even if there is an unreachable(); */
+export function defaultTypeValue(nativeType: NativeType, module: Module): ExpressionRef {
+  switch (nativeType) {
```

Comment:
There's already a `makeZero` helper - wondering if that'd be sufficient already?

Reply:
Yay this is exactly what I wanted!